### PR TITLE
Design Picker: Separate it into anchor-fm and site-setup design picker

### PR DESCRIPTION
**Pull request by arthur791004**
#### Proposed Changes

* As site-setup design picker and anchor-fm design picker are mixed up and are easily broken, I make a change to separate it into anchor-fm and site-setup design picker and then they won't affect each other.

#### Testing Instructions

**Site Setup Flow**

* Go to `/setup/vertical?siteSlug=<your_site>`
* Select a vertical
* Select a build intent
* Make sure the generated design picker works well
* Click the "View more options" and make sure the static design picker works well

**Anchor.fm flow**

* Go to `/setup/?anchor_podcast=9370e048`
* Finish the flow and make sure design picker works well

Related to https://github.com/Automattic/wp-calypso/pull/64402#issuecomment-1155311226

### CODE REVIEW

**Comment by arthur791004** [comment]
@sixhours I'm sorry for breaking Anchor.fm Design Picker in https://github.com/Automattic/wp-calypso/pull/64402 again, so I make a change to use `DesignSetup` component as an entry point to decide which design picker should be rendered by the flow. So, the Anchor.fm flow won't land on `SiteSetupDesignPicker` anymore and we don't need to handle all of logic in a single file. What do you think?

File: client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/site-setup-design-picker.tsx [hunk 1]
```diff
@@ -0,0 +1,457 @@
...
+	const heading = (
+		<FormattedHeader
+			id={ 'step-header' }
+			headerText={ headerText() }
+			subHeaderText={ subHeaderText() }
+			align="left"
+		/>
+	);
+
+	const stepContent = showGeneratedDesigns ? (
```
Comment (miksansegundo) [inline client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/site-setup-design-picker.tsx:362]: What do you think about these methods to remove the large ternary condition from the `stepContent` constant?

**Method 1**
```
let stepContent = (
  <DesignPicker
  ...
  />
)
if ( showGeneratedDesigns ) stepContent = (
  <GeneratedDesignPicker
  ...
  />
)
<DesignPicker
  stepContent={ stepContent }
/>
```

**Method 2**
```
const designPicker = (
  <DesignPicker
  ...
  />
)
const generatedDesignPicker = (
  <GeneratedDesignPicker
  ...
  />
)
<DesignPicker
  stepContent={ showGeneratedDesigns ? generatedDesignPicker : designPicker }
/>
```
These methods or even others could make the code cleaner thanks to removing the large ternary

**Review by miksansegundo — APPROVED** [review]
I have tested both paths and worked fine 👍 

Kudos for making this change because it makes that part much easier to read  🙌

**Review by sixhours — APPROVED** [review]
Yay, this works a treat. Thank you for this refactor, it will make working with the Anchor flow so much easier. :)

File: client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/site-setup-design-picker.tsx (hunk 1 above)
Comment (arthur791004) [reply client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/site-setup-design-picker.tsx:362]: I still prefer the current way but it's just my personal preference. I think maybe creating a function to render the content might be better. In further, I think we could also separate them into different components and share re-usable functions.

**Pull request merged**